x-pack/plugins/apm/server/lib/errors/distribution/get_buckets.ts

```diff
@@ -67,10 +67,12 @@ export async function getBuckets({
 
   const resp = await client.search(params);
 
+  const totalHits = resp.hits.total.value;
+
   const buckets = (resp.aggregations?.distribution.buckets || []).map(
     (bucket) => ({
       key: bucket.key,
-      count: bucket.doc_count,
+      count: totalHits === 0 ? undefined : bucket.doc_count,
```

Comment:
Doesn't it make more sense to have a top-level flag that indicates no hits? Seems kind of wasteful (and unclear) to have to iterate through the buckets to determine whether there are no hits. Not sure how we do this in other charts though, IIRC it's kind of a mess.

Reply:
Ok, I change it back to how it was and handled the empty state based on the `noHits` value.